### PR TITLE
Module definition update

### PR DESCRIPTION
**Pull request by agudek**
- Seperated svg and axis creation from module body. Visual parameters can be set through defining fields.

- Axes can be scaled to fit received data. The scaling can be set by defining fields.

- Removed scatter modulelist.

- Removed x-axis of pull-request module as it was not useful.

See default-module-values.js for the fields to define. Documentation for new module definition will follow.

### CODE REVIEW

File: src/modules/average-comment-size-yours.js [hunk 1]
```diff
@@ -34,73 +78,21 @@ define(function () {
...
-            var svg = d3.select(document.createElementNS(d3.ns.prefix.svg, 'svg'))
-                .attr("width", '100%')
-                .attr("height", '100%')
-                .attr("viewBox", "0 0 "+w+" "+h);
-
             var maxValue = Math.max.apply(Math,sizeData.map(function(o){return o.y;}));
 
             var xSizeScale = d3.scale.linear().domain([0,sizeData.length]).range([pad,w-17]),
             ySizeScale = d3.scale.linear().domain([maxValue,0]).range([padTop, h-padBottom]).nice();
+            var g = d3.select(document.createElementNS(d3.ns.prefix.svg, "g"));
```
Comment (borek2) [inline src/modules/average-comment-size-yours.js]: Naming of variable not very clear here

File: src/modules/pr-size.js [hunk 2]
```diff
@@ -34,92 +78,35 @@ define(function () {
...
-                .attr("y2",h-padBottom)
-                .attr("style","stroke-width:1px;stroke:black");
-
-            svg.append("line")
-                .attr("x1",pad)
-                .attr("x2",pad)
-                .attr("y1",h-padBottom)
-                .attr("y2",0)
-                .attr("style","stroke-width:1px;stroke:black");
+            var g = d3.select(document.createElementNS(d3.ns.prefix.svg, "g"));
```
Comment (borek2) [inline src/modules/pr-size.js]: again ;)

File: src/svgCreator.js [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        
+        //var rect = g.append("rect");
+
+        //var text = 
+        g.append("text")
+            .attr("x",720/2)
+            .attr("y",360)
+            .attr("text-anchor", "middle")
+            .text(label);
+/*
```
Comment (borek2) [inline src/svgCreator.js]: remove commented lines

**Comment by borek2** [comment]
Again really nice work !! The naming of the graph files should be improved though, sometimes its called graph1.js, other time its the real name of the graph. We might want to put the graphs into a different folder so we can retrieve them easily?

File: src/svgCreator.js (hunk 3 above)
Comment (agudek) [inline src/svgCreator.js]: I am going to add a rectangle behind the axis labels in a future sprint, so would like to keep this code for now :)

File: src/modules/average-comment-size-yours.js (hunk 1 above)
Comment (agudek) [inline src/modules/average-comment-size-yours.js]: It's literally an encapsulatin <g></g> tag. Did not think it would need a special name.

**Comment by mboom** [comment]
Can you give a short description about the (new) structure, how files and classes cooperate with each other?

**Comment by agudek** [comment]
Will update the module definition document soon

**Pull request merged**